src/jdbc/resultset.clj

```diff
@@ -52,9 +58,12 @@
                (when (.next rs)
                  (cons (values) (lazy-seq (thisfn)))))
         records (fn thisfn []
-                  (when (.next rs)
-                    (-> (zipmap keyseq (values))
-                        (cons (lazy-seq (thisfn))))))]
+                  (let [keyseq (if metadata->identifiers
+                                 (metadata->identifiers metadata)
+                                 (mapv (comp keyword identifiers (fn [^long i] (.getColumnLabel metadata i))) idseq))]
+                    (when (.next rs)
```

Comment:
I think putting the keyseq inside the lazyseq is a wrong approach because it not reuses it and continuosly creates a new in each itearion. I'll try to understand this problem an properly fix it.